torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -55,6 +161,52 @@ bool IRInputOutput::hasOutput(const Val* const output) const {
   return false;
 }
 
+void IRInputOutput::replaceInput(Val* replace, Val* with) {
+  bool changed = false;
+  for (decltype(inputs_.size()) i{0}; i < inputs_.size(); i++) {
+    if (inputs_[i] == replace) {
+      inputs_[i] = with;
+      changed = true;
+      break;
+    }
+  }
+  TORCH_INTERNAL_ASSERT(
+      changed,
+      "Error detected when trying to replace input ",
+      replace,
+      " with ",
+      with,
+      " .");
+}
+
+void IRInputOutput::replaceOutput(Val* replace, Val* with) {
+  bool changed = false;
+  for (decltype(outputs_.size()) i{0}; i < outputs_.size(); i++) {
+    if (outputs_[i] == replace) {
+      outputs_[i] = with;
+      changed = true;
+      break;
+    }
+  }
+  TORCH_INTERNAL_ASSERT(
+      changed,
+      "Error detected when trying to replace output ",
+      replace,
+      " with ",
+      with,
+      " .");
+}
+
+void IRInputOutput::removeInput(Val* val) {
+  auto it = inputs_.begin();
+  for (; it != inputs_.end(); ++it) {
+    if ((*it) == val)
+      break;
+  }
+  assert(it != inputs_.end());
```

Comment:
I had few instances where assert is not doing the right thing for me. Maybe we should switch to `TORCH_INTERNAL_ASSERT`

Reply:
changed